### PR TITLE
HID-1827: Common Design - Footer

### PR DESCRIPTION
**Pull request by rupl**
## https://humanitarian.atlassian.net/browse/HID-1827

Implementing the Common Design footer on HID. The branch actually includes the entirety of the CD but only modifies the markup of the Footer to use the new assets.

The work actually introduces some significant regressions to Sidebar functionality, but [they've been ticketed](https://humanitarian.atlassian.net/browse/HID-1831).

### CODE REVIEW

**Comment by rupl** [comment]
E2E reported 4 more failures than "normal" — I won't fix them because we know there are regressions, but also a good comparative number to have when I start fixing again.

> ca5c1e9: Executed 106 of 106 specs (5 ERRORS) (75 FAILED) in 15 mins 43 secs.

**Comment by rupl** [comment]
@left23 adding you as a reviewer, but please do NOT feel obligated set this repo up. Too much adventure for one review. A quick overview of code will be enough and you can see how I deviated from standard CD in the last couple commits.

File: src/app/components/footer/footer.html [hunk 1]
```diff
@@ -1,49 +1,70 @@
...
...a2.37 2.37 0 0 0 2.2-2.51v-9A2.37 2.37 0 0 0 45.8 34zM36 43a2 2 0 1 1 2-2 2 2 0 0 1-2 2zm7 0a2 2 0 1 1 2-2 2 2 0 0 1-2 2z"/></symbol><symbol viewBox="0 0 48 48" id="user" xmlns="http://www.w3.org/2000/svg"><path d="M23 24h2a16 16 0 0 1 16 16v4H7v-4a16 16 0 0 1 16-16z"/><rect x="15" y="4" width="18" height="18" rx="9" ry="9"/></symbol><symbol viewBox="0 0 48 48" id="user--white" xmlns="http://www.w3.org/2000/svg"><path fill="#fff" d="M23 24h2a16 16 0 0 1 16 16v4H7v-4a16 16 0 0 1 16-16z"/><rect fill="#fff" x="15" y="4" width="18" height="18" rx="9" ry="9"/></symbol><symbol viewBox="0 0 48 48" id="users" xmlns="http://www.w3.org/2000/svg"><rect x="9" y="7" width="16" height="16" rx="8" ry="8"/><rect x="27" y="11" width="12" height="12" rx="6" ry="6"/><path d="M34 25h-2a11 11 0 0 0-3.8.69A16 16 0 0 1 34 38v1h11v-3a11 11 0 0 0-11-11zM16 25h2a13 13 0 0 1 13 13v3H3v-3a13 13 0 0 1 13-13z"/></symbol><symbol viewBox="0 0 48 48" id="warning" xmlns="http://www.w3.org/2000/svg"><path d="M37 2H11a8 8 0 0 0-8 8v19a8 8 0 0 0 8 8h14l8.8 8.81a.7.7 0 0 0 1.2-.51V37h2a8 8 0 0 0 8-8V10a8 8 0 0 0-8-8zm-15 8a2 2 0 1 1 4 0v11a2 2 0 0 1-4 0zm2 21a2.5 2.5 0 1 1 2.5-2.5A2.5 2.5 0 0 1 24 31z"/></symbol></svg>
```
Comment (rupl) [inline src/app/components/footer/footer.html:70]: @left23 I noticed that adding the SVG (regardless of where or how) caused a collision in this codebase. The SVG contains a `<symbol id="search">` and we also have an input with `id="search"` in the UI.

I don't want to change the CD sprite right now, but wanted to highlight this as something we might want to change in the future by namespacing our HTML IDs, particularly when they are so generic and common like "search" or "back" etc.

More info: https://humanitarian.atlassian.net/browse/HID-1832

Comment (left23) [reply src/app/components/footer/footer.html:70]: Well spotted. The IDs are coming from the SVG file names but there are config options for prefixing we should avail of https://github.com/jkphl/svg-sprite#common-mode-properties

**Review by left23 — APPROVED** [review]
Thanks for pinging me on this. I'm always curious to see how things get implemented.

File: src/app/components/footer/footer.html (hunk 1 above)
Comment (rupl) [reply src/app/components/footer/footer.html:70]: ooh interesting that we could potentially prefix per-project. For now I've simply done a compile within the canonical `ocha_basic` project and copied the assets over.

However it's not that much extra work to compile with a prefix and re-copy assets (along with instructions in the README of course!). Thanks for pointing me to the spriting docs!

**Comment by rupl** [comment]
After messing with the duplicate HTML ID for too long, I'm going to leave it for now and file a ticket to specifically clean that regression up. In the mean time I'm not ever seeing malfunctions due to the duplicate IDs so I'll merge this in.

_Edit:_ I already filed it https://humanitarian.atlassian.net/browse/HID-1832

**Pull request merged**